pkg/providers/aws/provider_postgres.go

```diff
@@ -965,6 +969,30 @@ func (p *PostgresProvider) exposePostgresMetrics(ctx context.Context, cr *v1alph
 	}
 }
 
+// set metrics about the postgres instance being deleted
+// works in a similar way to kube_pod_deletion_timestamp
+// https://github.com/kubernetes/kube-state-metrics/blob/0bfc2981f9c281c78e33052abdc2d621630562b9/internal/store/pod.go#L200-L218
+func (p *PostgresProvider) setPostgresDeletionTimestampMetric(ctx context.Context, cr *v1alpha1.Postgres) {
+	if cr.DeletionTimestamp != nil && !cr.DeletionTimestamp.IsZero() {
+		// build instance name
+		instanceName, err := p.buildInstanceName(ctx, cr)
+		if err != nil {
+			logrus.Errorf("error occurred while building instance name during postgres metrics: %v", err)
+		}
+
+		// get Cluster Id
+		logrus.Info("setting postgres information metric")
+		clusterID, err := resources.GetClusterID(ctx, p.Client)
+		if err != nil {
+			logrus.Errorf("failed to get cluster id while exposing information metric for %v", instanceName)
+			return
+		}
+
+		labels := buildPostgresStatusMetricsLabels(cr, clusterID, instanceName, cr.Status.Phase)
```

Comment:
Using the status metric would mean there could be potentially multiple metrics exposed for the one resource as the phase changes. 

Would it be worth a single metric using the build generic label function. 

We can build an alert on the basis of if we have a present `cro_deletion_timestamp` metric and a `cro_phase_failed` metric = 1

Reply:
Yeah I guess so. I'm not really sure to be honest 😅

My thinking was more along the lines of an alert could look for a `cro_deletion_timestamp` metric that included the label `statusPhase="failed"` which is what will happen with this implementation, but maybe I don't understand how to build alerts.

Reply:
Had a call with @david-martin and he mentioned that the query `cro_postgres_deletion_timestamp{statusPhase="failed"}` should work.

Reply:
cool cool, i wasn't 100% sure either :smile_cat: